packages/api/tests/api/judge/post.test.ts

```diff
@@ -0,0 +1,87 @@
+import { Judge } from '@hangar/database';
+import { post } from '../../../src/api/judge/post';
+import { createMockRequest } from '../../testUtils/expressHelpers/createMockRequest';
+import { createMockResponse } from '../../testUtils/expressHelpers/createMockResponse';
+import { getMock } from '../../testUtils/getMock';
+import { validatePayload } from '../../../src/utils/validatePayload';
+import { EntityManager } from '@mikro-orm/knex';
+
+jest.mock('@hangar/database', () => ({
+  Judge: jest.fn(),
+}));
+
+jest.mock('../../../src/utils/validatePayload');
+const validatePayloadMock = getMock(validatePayload);
+
+describe('judge post endpoint', () => {
+  it('bails if a validation error occurs', async () => {
+    validatePayloadMock.mockReturnValueOnce({ errorHandled: true });
+
+    const req = createMockRequest({ query: { inviteCode: 'xyz' } });
+    const res = createMockResponse();
+
+    await post(req as any, res as any);
+
+    expect(validatePayloadMock).toBeCalledTimes(1);
+  });
+
+  it('should return 403 for an invalid inviteCode', async () => {
+    validatePayloadMock.mockReturnValueOnce({
+      errorHandled: false,
+      data: { inviteCode: 'xyz' },
+    } as any);
+
+    const req = createMockRequest({ query: { inviteCode: 'xyz' } });
+    const res = createMockResponse();
+
```

Comment:
Make sure to mock the return value of `findOne` to return null, otherwise I believe this test is just passing accidentally